presto-native-execution/scripts/setup-conda.sh

```diff
@@ -0,0 +1,177 @@
+      if [[ "${input}" == "Y" || "${input}" == "y" || "${input}" == "" ]]; then
+        return 0
+      elif [[ "${input}" == "N" || "${input}" == "n" ]]; then
+        return 1
+      fi
+    done
+  ) 2> /dev/null
+}
+
+# Copied from velox/scripts/setup-ubuntu.sh
+function install_fmt {
+  github_checkout fmtlib/fmt 8.0.1
+  cmake_install -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" -DFMT_TEST=OFF
+}
+
+# Copied from velox/scripts/setup-ubuntu.sh
+function install_folly {
+  github_checkout facebook/folly "${FB_OS_VERSION}"
+  cmake_install -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" -DBUILD_TESTS=OFF -DFOLLY_HAVE_INT128_T=ON
+}
+
+# Copied from velox/scripts/setup-ubuntu.sh
+function install_fizz {
+  github_checkout facebookincubator/fizz "${FB_OS_VERSION}"
+  cmake_install -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" -DBUILD_TESTS=OFF -S fizz
+}
+
+# Copied from velox/scripts/setup-ubuntu.sh
+function install_wangle {
+  github_checkout facebook/wangle "${FB_OS_VERSION}"
+  cmake_install -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" -DBUILD_TESTS=OFF -S wangle
+}
+
+# Copied from velox/scripts/setup-ubuntu.sh
+function install_fbthrift {
+  github_checkout facebook/fbthrift "${FB_OS_VERSION}"
+  cmake_install -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" -DBUILD_TESTS=OFF
+}
+
+function install_six {
+  pip3 install six
+}
+
+function install_proxygen {
+  github_checkout facebook/proxygen "${FB_OS_VERSION}"
+  cmake_install -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" -DBUILD_TESTS=OFF
+}
+
+function install_antlr4 {
+  cd "${DEPENDENCY_DIR}"
+    if [ -d "antlr4-cpp-runtime-4.9.3-source" ]; then
+      rm -rf antlr4-cpp-runtime-4.9.3-source
+    fi
+  wget https://www.antlr.org/download/antlr4-cpp-runtime-4.9.3-source.zip \
+    -O antlr4-cpp-runtime-4.9.3-source.zip
+  mkdir antlr4-cpp-runtime-4.9.3-source && cd antlr4-cpp-runtime-4.9.3-source
+  unzip ../antlr4-cpp-runtime-4.9.3-source.zip
+  mkdir build && mkdir run && cd build
+  cmake -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" .. && make "-j${NPROC}" install
+}
+
+function install_system_deps {
+  conda install --yes -c conda-forge cmake ninja boost \
+    double-conversion gflags glog libevent zlib lz4 snappy c-ares \
+    openssl icu gmock zstd re2 libsodium libthrift \
+    bison flex tzdata libunwind gperf unzip make pkg-config \
+    cxx-compiler kernel-headers_linux-64 lzo
+}
+
+# Copied from velox/scripts/setup-ubuntu.sh
+function install_velox_deps {
+  echo "Installing velox dependencies"
+  run_and_time install_fmt
+  run_and_time install_folly
+  run_and_time install_fizz
+  run_and_time install_wangle
+  run_and_time install_fbthrift
+}
+
+function install_presto_deps {
+  echo "Installing presto dependencies"
+  run_and_time install_six
+  run_and_time install_proxygen
+  run_and_time install_antlr4
+}
+
+function print_summary {
+  set +x
+  echo "All done!"
+  echo "To enter conda environment, restart your shell and run:"
+  echo "source $BASH_SOURCE"
+  echo "conda_activate_env"
+}
```

Comment:
Move function calls from the ubuntu script here and add a check whether this file is sourced (then don't execute the commands)